metagpt/tools/sd_engine.py

```diff
@@ -62,52 +62,54 @@ def __init__(self):
         # Define default payload settings for SD API
         self.payload = payload
         logger.info(self.sd_t2i_url)
-
+    
     def construct_payload(
-        self,
-        prompt,
-        negtive_prompt=default_negative_prompt,
-        width=512,
-        height=512,
-        sd_model="galaxytimemachinesGTM_photoV20",
+            self,
+            prompt,
+            negative_prompt=default_negative_prompt,
+            width=512,
+            height=512,
+            sd_model="galaxytimemachinesGTM_photoV20",
+            **kwargs
     ):
         # Configure the payload with provided inputs
         self.payload["prompt"] = prompt
-        self.payload["negtive_prompt"] = negtive_prompt
+        self.payload["negative_prompt"] = negative_prompt
         self.payload["width"] = width
         self.payload["height"] = height
         self.payload["override_settings"]["sd_model_checkpoint"] = sd_model
+        self.payload.update(**kwargs)
         logger.info(f"call sd payload is {self.payload}")
         return self.payload
-
+    
     def _save(self, imgs, save_name=""):
         save_dir = WORKSPACE_ROOT / "resources" / "SD_Output"
         if not os.path.exists(save_dir):
```

Comment:
The `save_dir` path is constructed using the Path constructor. You can replace it with the following code:
```python
if not save_dir.exists():
    mkdir(parents=True, exist_ok=True)
```